dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java

```diff
@@ -349,59 +375,59 @@ public List<IHTMLPage> getHTMLPagesByContainer(final String containerId) throws
     }
 
     @CloseDBIfOpened
-	@Override
-	public List<IHTMLPage> getHTMLPages(Object parent, boolean live,
-			boolean deleted, int limit, int offset, String sortBy, User user,
-			boolean respectFrontEndRoles) throws DotDataException,
-			DotSecurityException {
-		List<IHTMLPage> pages = new ArrayList<>();
-		StringBuffer query = new StringBuffer();
-		String liveWorkingDeleted = (live) ? " +live:true "
-				: (deleted) ? " +working:true +deleted:true "
-						: " +working:true -deleted:true";
-		query.append(liveWorkingDeleted);
-		if (parent instanceof Folder) {
-			query.append(" +conFolder:" + ((Folder) parent).getInode());
-			query.append(" +conHost:" + ((Folder) parent).getHostId());
-		} else if (parent instanceof Host) {
-			query.append(" +conFolder:SYSTEM_FOLDER +conHost:"
-					+ ((Host) parent).getIdentifier());
-		// if not a folder or host the filtering is done by template (parent) 
-		}else if (parent instanceof String)
-			if(!((String)parent).isEmpty()){
-				// list of content types (htmlpage type)
-				List<Structure> structures = StructureFactory.getStructures("structureType="+Structure.STRUCTURE_TYPE_HTMLPAGE, "", 0, 0, "");
-				StringBuilder structuresList = new StringBuilder();
-				boolean notOR = true;
-				
-				// creates a list of content types with the template field e.g. htmlpageasset.template:## OR newpages.template:##
-				for(Structure structure: structures){
-					if(notOR){
-						notOR=!notOR;
-					}else
-						structuresList.append(" OR ");
-					structuresList.append(structure.getVelocityVarName());
-					structuresList.append(".template:");
-					structuresList.append((String)parent);					
-				}
-				if(structuresList.length()>0)
-					query.append(" +("+ structuresList.toString().trim()+")" );
-			}
-		
-		query.append(" +structureType:" + Structure.STRUCTURE_TYPE_HTMLPAGE);
-		if (!UtilMethods.isSet(sortBy)) {
-			sortBy = "modDate asc";
-		}
-		List<Contentlet> contentlets = contentletAPI.search(
-				query.toString(), limit, offset, sortBy, user,
-				respectFrontEndRoles);
-		for (Contentlet cont : contentlets) {
-			if(UtilMethods.isSet(fromContentlet(cont).getInode()))
-				pages.add(fromContentlet(cont));
-		}
-		return pages;
-	}
-    
+    @Override
+    public List<IHTMLPage> getHTMLPages(Object parent, boolean live,
```

Comment:
I see you are calling a factory here, double check if CloseInDbIfOpened you be in this method